patternfly/patternfly-elements
# fix: Typography mixin arguments & update docs

## <Pull request description>

- Add support to the $base argument so that it works regardless if the $type argument is "title" or "text". Right now it does not work if the $type is title.
- Updates pfe-clipboard component so that it leverages system typography variables
- Add alias mixin for easier use within components:  `@mixin pfe-c-typography`

### Related issues

- https://github.com/patternfly/patternfly-elements/issues/1384


### Preview
 
Link(s) to demo page(s) where this element can be viewed:
- [pfe-styles](https://deploy-preview-1385--happy-galileo-ea79c4.netlify.app/elements/pfe-styles/demo/)
- [pfe-clipboard](https://deploy-preview-1385--happy-galileo-ea79c4.netlify.app/elements/pfe-clipboard/demo/)
- [pfe-sass docs](https://deploy-preview-1385--happy-galileo-ea79c4.netlify.app/elements/pfe-sass/demo/#typography-mixin-pfe-typography)

### Sandbox links

 - https://web-dev-drupal-webux3.dev.a1.vary.redhat.com/en/demo-web-components#classes
 - https://web-dev-drupal-webux3.dev.a1.vary.redhat.com/en/potter-long-form-article-test

### What has changed and why



- Modified pfe-typography mixin to universally apply $base argument to title & text types
- Added alias mixin to make it easier for components to just get the font-size they need


### Testing instructions

Nothing should regress, because the PFE project is not setting base to false (its more useful for other projects like webrh). You can verify a few pages like [pfe-band](https://deploy-preview-1385--happy-galileo-ea79c4.netlify.app/elements/pfe-band/demo/) - they should look the same

If you are using the `pfe-typography` function, you should be able to opt out of the extra properties (beyond font-family, font-size) if desired, via the `$base` argument set to false:

Test by throwing this sass into some .scss file in pfe-styles:

```scss
.foo {
  @extend %pfe-title--lg;
}
.bar {
 @include pfe-typography(lg, $type: "title"); 
}
.baz {
 @include pfe-typography(lg, $type: "title", $base: false); 
}
```

the expected output should return:

```css
.foo {
  font-family: "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif;
  font-family: var(--pfe-theme--font-family--heading, "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif);
  font-size: 18px;
  font-size: var(--pf-c-title--m-lg--FontSize, var(--pf-global--FontSize--lg, 18px));
  line-height: 1.5;
  line-height: var(--pfe-theme--line-height, 1.5);
  font-weight: 400;
  font-weight: var(--pfe-theme--font-weight--normal, 400);
}

.foo:not(:last-child) {
  margin-bottom: 16px;
  margin-bottom: var(--pfe-theme--content-spacer--heading--sm, 16px);
}

.bar {
  font-family: "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif;
  font-family: var(--pfe-theme--font-family--heading, "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif);
  font-size: 18px;
  font-size: var(--pf-c-title--m-lg--FontSize, var(--pf-global--FontSize--lg, 18px));
  line-height: 1.5;
  line-height: var(--pfe-theme--line-height, 1.5);
  font-weight: 400;
  font-weight: var(--pfe-theme--font-weight--normal, 400);
}

.bar:not(:last-child) {
  margin-bottom: 16px;
  margin-bottom: var(--pfe-theme--content-spacer--heading--sm, 16px);
}

.baz {
  font-family: "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif;
  font-family: var(--pfe-theme--font-family--heading, "Red Hat Display", "RedHatDisplay", "Overpass", Overpass, Arial, sans-serif);
  font-size: 18px;
  font-size: var(--pf-c-title--m-lg--FontSize, var(--pf-global--FontSize--lg, 18px));
  line-height: 1.5;  /* This shouldn't be here */
  line-height: var(--pfe-theme--line-height, 1.5); /* This shouldn't be here */
  font-weight: 400; /* This shouldn't be here */
  font-weight: var(--pfe-theme--font-weight--normal, 400); /* This shouldn't be here */
}

.baz:not(:last-child) {
  margin-bottom: 16px; /* This shouldn't be here */
  margin-bottom: var(--pfe-theme--content-spacer--heading--sm, 16px); /* This shouldn't be here */
}
```
 
## Full list of things you can throw into the mixin

- http://pastebin.test.redhat.com/947129


### Ready-for-merge Checklist



- n/a Tests have been updated to cover these changes.
- n/a Browser testing passed.
- [x] Documentation (README.md, WHY.md, etc.) updated or added:  https://github.com/patternfly/patternfly-elements/pull/1395 
- n/a Approved by designer.


### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
